File: jcarrollonlinev4.backend/Models/Blog/BlogFeedItemModel.cs
using jcarrollonlinev4.backend.Models.Users;
using System.ComponentModel.DataAnnotations;

namespace jcarrollonlinev4.backend.Models.Blog
{
    public class BlogFeedItemModel : BlogFeedModelBase
    {
        public BlogFeedItemModel()
        {
            Author = new ApplicationUserModel();
            Comments = new BlogCommentsModel();
        }

        public int Id { get; set; }
        public ApplicationUserModel Author { get; set; }

        public string? AuthorId { get; set; }

        [Display(Name = "Title")]
        public string? Title { get; set; }

        [Display(Name = "Content")]
        [DataType(DataType.MultilineText)]
        //[AllowHtml]
        public string? Content { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public BlogCommentsModel Comments { get; set; }
    }
}

File: jcarrollonlinev4.frontend/src/models/Blog/BlogFeedItemModel.ts
import { ApplicationUserModel } from "../Users/ApplicationUserModel";
import { BlogCommentsModel } from "./BlogCommentsModel";
import { BlogFeedModelBase } from "./BlogFeedModelBase";

export interface BlogFeedItemModel extends BlogFeedModelBase {
  Id: number;
  Author?: ApplicationUserModel;
  AuthorId?: string;
  Title?: string;
  Content?: string;
  CreatedAt: Date;
  UpdatedAt: Date;
  Comments?: BlogCommentsModel;
}
